packages/xlayers/src/app/editor/preview-badge.service.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { TestBed } from '@angular/core/testing';
+import { WINDOW } from '@app/core/window.service';
+import { PreviewBadgeService } from './preview-badge.service';
+
+describe('PreviewBadgeService', () => {
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            providers: [
+                PreviewBadgeService,
+                {
+                    provide: WINDOW,
+                    useValue: {
+                    }
+                }
+            ]
+        });
+    });
+
+    it('should set badge to LOCAL PREVIEW when running on a local machine', () => {
+        TestBed.overrideProvider(WINDOW, { useValue: { location: { hostname: 'localhost' } } });
+        const service: PreviewBadgeService = TestBed.get(PreviewBadgeService);
```

Comment:
this line could be moved the beforeEach block to reduce duplication.

Reply:
Will fix this later today

Reply:
Maybe I'm missing something but when I move the `service` to the `beforeEach` the `Tokens` are not updated. I believe this is because when I access the service it will provide the token at the moment and that is in all cases a empty object. 

For this reason it will not have updated the token as expected and the tests will fail. Does this make sense?

Reply:
okay. leave it as it is. Will figure it out later.